### PR TITLE
Improving Password Security blogpost

### PR DESCRIPTION
**Pull request by alecmroo**
New blogpost on password security

### CODE REVIEW

**Comment by mvalipour** [comment]
Looks great!
Does it worth clarifying the method we used to determine someone's password were leaked before?

**Comment by alecmroo** [comment]
That's a great question! Getting into detail of how this works is a topic which has been dealt with extensively in both Facebook's and Troy Hunt's blogposts around the topics, e.g.:

* https://www.facebook.com/notes/protect-the-graph/keeping-passwords-secure/1519937431579736/
* https://www.troyhunt.com/inside-the-massive-711-million-record-onliner-spambot-dump/

…so, much as I might like to expand, I don't feel that writing it up would add much to the collective wisdom of humankind. 

The story here is that this is the right thing to do, because the means are already well-documented. :-)

**Comment by mvalipour** [comment]
Yep I understand that, but I think not all the readers would know that and automatically wonder where it comes from. So IMPOV exact same comment above may deserve to be mentioned.

File: _posts/2017-09-05-improving-password-security.md [hunk 1]
```diff
@@ -0,0 +1,49 @@
+---
+layout: post
+title:  "Improving Password Security"
+author: "Alec Muffett"
+excerpt: >
```
Comment (kapowaz) [inline _posts/2017-09-05-improving-password-security.md]: The excerpt here is quite short, which means it looks a little brief, as well as not having as much useful information indexable from the homepage:

img

I think it’d be straightforward to rejig the first two paragraphs to fix this:

```markdown
excerpt: >
  Password-reuse puts customer account security at risk when 
  third-party websites get popped. We want our customers to be safe
  online, and we --- specifically the Deliveroo  Infrastructure
  Security team --- want to better protect our customers' accounts.
---

We're starting from a good place: our passwords are hashed using 
the `bcrypt` algorithm, a robust and industry-standard password hash.
The Rails default for `bcrypt` is to run at strength 10, meaning 
2<sup>10</sup> = 1024 rounds of hashing, a reasonable work factor for
a modern password hash.
```

This ends up looking like this:

(homepage)

img

(article)

img

File: _posts/2017-09-05-improving-password-security.md [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+_"popped"_ --- bad people hack into them and exfiltrate password data,
+often sharing their findings with the world through pastebin sites and
+bulletin-boards.
+
+These actions put at risk any site where the owner has reused the same
+login name and password.
+
+There are excellent free tools which our customers may use to help
+discover if they are at risk --- for instance Troy Hunt's 
+[Have I Been Pwned?](https://haveibeenpwned.com/) website; and we
```
Comment (kapowaz) [inline _posts/2017-09-05-improving-password-security.md]: Quibble, but should that be semicolon be a comma?

Comment (alecmroo) [reply _posts/2017-09-05-improving-password-security.md]: I think not, we move from a specific to a more general comment.

Comment (kapowaz) [reply _posts/2017-09-05-improving-password-security.md]: It’s the way it’s followed by `and` that looks odd to me — I’d say it should either be `, and we` or `; we`, but it’s not a particularly big deal 😁

**Pull request merged**